src/cube_game.py

```diff
@@ -0,0 +1,98 @@
+from functools import reduce
+from operator import mul
+
+
+class CubeGame:
+    def __init__(self, id: int, reveals: list[dict[str, int]]):
+        self.id = id
+        self.reveals = reveals
+
+        self.min_colors: dict[str, int] = {"red": 0, "green": 0, "blue": 0}
+
+        for reveal in reveals:
+            for key, val in reveal.items():
+                if self.min_colors[key] < val:
+                    self.min_colors[key] = val
+
+    @staticmethod
+    def from_string(data: str) -> "CubeGame":
```

Comment:
It's times like these that I don't get why Python is propped up as a great language, and at the same time, JavaScript gets such a terrible rap.

Let's be honest, can we all agree that having to put "CubeGame" in quotes here is just garbage? And honestly, the `@staticmethod` attribute just hurts me to look at... If the language requires you to pass in `self` as the first argument of an instance method, it should know that if you don't pass in `self`, it's a static method.